File: app/src/main/java/com/bogdan/codeforceswatcher/util/AnalyticsController.kt
package com.bogdan.codeforceswatcher.util

import android.os.Bundle
import com.bogdan.codeforceswatcher.CwApp
import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.crashlytics.FirebaseCrashlytics
import io.xorum.codeforceswatcher.util.IAnalyticsController

class AnalyticsController : IAnalyticsController {

    private val instance = FirebaseAnalytics.getInstance(CwApp.app)

    override fun logError(message: String) {
        FirebaseCrashlytics.getInstance().recordException(Throwable(message))
    }

    override fun logEvent(eventName: String, params: Map<String, String>) {
        instance.logEvent(eventName, params.toBundle())
    }

    private fun Map<String, String>.toBundle() = Bundle().apply {
        for (entry in entries)
            putString(entry.key, entry.value)
    }
}

File: ios/Codeforces Watcher/Application/utils/AnalyticsController.swift
//
//  AnalyticsController.swift
//  Codeforces Watcher
//
//  Created by Den Matyash on 07.09.2020.
//  Copyright © 2020 xorum.io. All rights reserved.
//

import Foundation
import common
import FirebaseAnalytics
import FirebaseCrashlytics

class AnalyticsController: IAnalyticsController {
    
    func logError(message: String) {
        Crashlytics.crashlytics().record(error: CrashlyticsError(message: message))
    }
    
    func logEvent(eventName: String, params: [String: String] = [:]) {
        Analytics.logEvent(eventName, parameters: params)
    }
}

